### PR TITLE
create controller tests

### PR DESCRIPTION
**Pull request by gdumani**
verify pages rendered on index and show methods for users and posts controllers

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 
-  describe 'GET /show' do
+  describe 'GET /show retrieve place holder page' do
     it 'returns http success' do
-      get '/posts/show'
+      get "/users/#{@usr.id}/posts/#{@pst.id}"
       expect(response).to have_http_status(:success)
+      expect(response.body.include?('Given post of given user')).to be true
     end
   end
```
Comment (cynthiainga) [inline spec/requests/posts_spec.rb]: - [x] Spec file should include tests that ensure the correct view template is rendered.
Example: `expect(response).to render_template(:show)`, please kindly add for `show` and `index` as well 🙏🏾

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
-  describe 'GET /show' do
+  describe 'GET /show and render place holder page' do
     it 'returns http success' do
-      get '/users/show'
+      @usr = User.create name: 'Testuser', photo: 'www.photo', bio: 'Test bio', posts_counter: 0
+      get "/users/#{@usr.id}"
       expect(response).to have_http_status(:success)
+      expect(response.body.include?('Given user with recent posts')).to be true
     end
   end
```
Comment (cynthiainga) [inline spec/requests/users_spec.rb]: - [ ] Spec file should include tests that ensure the correct view template is rendered.
Example: `expect(response).to render_template(:show)`, please kindly add for `show` and `index` as well 🙏🏾

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @gdumani, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Correct Gitflow ✔️ 
- [ ] Has and uses RSpec gem for testing ✔️ 
- [ ] Used `type: request` ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gdumani** [comment]
Added the suggested test. Kindly explain why is it needed since documentation on this page
https://rspec.info/features/6-0/rspec-rails/controller-specs/render-views/
indicates that checking the response.body is a way to evaluate the correctness of the template used.
Thanks

**Review by alexander16108 — APPROVED** [review]
Hi @gdumani , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Correct Gitflow ✔️
 Has and uses RSpec gem for testing ✔️
 Used type: request ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**